cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu.go

```diff
@@ -451,6 +472,118 @@ func createClusterDirectory(clusterName string) (string, error) {
 	return fp, nil
 }
 
+// Returns the latest TKr compatible image path:tag string
+// If none is found for version of CLI, returns an error
+func getLatestCompatibleTkr() (string, error) {
+	compatibilityFileName, err := getCompatibilityFile()
+	if err != nil {
+		return "", err
+	}
+
+	compatibilityDirPath, err := getUnmanagedCompatibilityPath()
+	if err != nil {
+		return "", err
+	}
+
+	// Read compatibility file, returns a compatibility struct
+	c, err := tkr.ReadCompatibilityFile(filepath.Join(compatibilityDirPath, compatibilityFileName))
+	if err != nil {
+		return "", err
+	}
+
+	// Inspect CLI version and get most recent compatible version of TKR
+	for _, cliVersion := range c.UnmanagedClusterPluginVersions {
+		if cliVersion.Version == plugin.Version {
+			// We've found a compatible version
+			// TODO - Just returning the first one. How do we handle multiple compatible TKrs?
+			return "projects.registry.vmware.com/tce/" + cliVersion.SupportedTkrVersions[0].ImagePath + ":" + cliVersion.SupportedTkrVersions[0].ImageTag, nil
+		}
+	}
+
+	return "", fmt.Errorf("could not find compatible CLI version in compatibility file")
```

Comment:
The way this bubbles up, it gives the user a bit of a confusing error message. Namely, it says:

![image](https://user-images.githubusercontent.com/6200057/159814083-1367f09d-2845-48b7-a19e-2190b9ea26c8.png)

Is there a way (without too much complex logic) we could tell the user:

"Failed parsing the TKr compatibility file: Error...."